api/src/test/java/org/openmrs/module/fgh/mpi/MpiHttpClientTest.java

```diff
@@ -1,11 +1,184 @@
+	
+	private static final String MESSAGE_HEADER_REFERENCE = "metadata.epts.e-saude.net/bundle";
+	
+	private static final String MESSAGE_HEADER_EVENT_URI = "urn:ihe:iti:pmir:2019:patient-feed";
+	
+	private static final AuthenticationType AUTHENTICATION_TYPE = AuthenticationType.OAUTH;
+	
+	private static final String MPI_BASE_URL = "http://sante.org.mz";
+	
+	private static final String MPI_APP_CONTENT_TYPE = "application/fhir+json";
+	
+	private static final MpiSystemType MPI_SYSTEM = MpiSystemType.SANTEMPI;
+	
+	private static final String SANTE_CLIENT_ID = "client_credentials";
+	
+	private static final String SANTE_CLIENT_SECRET = "bG6TuS3X-H1MsT4ctW!CxXjK9J4l1QpK8B0Q";
+	
+	@Rule
+	public ExpectedException expectedException = ExpectedException.none();
+	
+	@Mock
+	private MpiContext mpiContext;
+	
+	@Before
+	public void setup() {
+		PowerMockito.mockStatic(Context.class);
+		PowerMockito.mockStatic(MpiUtils.class);
+		PowerMockito.mockStatic(MpiContext.class);
+		PowerMockito.mockStatic(URL.class);
+		when(Context.getAdministrationService()).thenReturn(adminService);
+		when(adminService.getGlobalProperty(GP_AUTHENTICATION_TYPE)).thenReturn(AUTHENTICATION_TYPE.toString());
+		when(adminService.getGlobalProperty(GP_MPI_BASE_URL)).thenReturn(MPI_BASE_URL);
+		when(adminService.getGlobalProperty(GP_MPI_SYSTEM)).thenReturn(MPI_SYSTEM.toString());
+		when(adminService.getGlobalProperty(GP_MPI_APP_CONTENT_TYPE)).thenReturn(MPI_APP_CONTENT_TYPE);
+		when(adminService.getGlobalProperty(GP_SANTE_MESSAGE_HEADER_FOCUS_REFERENCE)).thenReturn(MESSAGE_HEADER_REFERENCE);
+		when(adminService.getGlobalProperty(GP_SANTE_MESSAGE_HEADER_EVENT_URI)).thenReturn(MESSAGE_HEADER_EVENT_URI);
+		when(adminService.getGlobalProperty(GP_SANTE_MESSAGE_HEADER_EVENT_URI)).thenReturn(MESSAGE_HEADER_EVENT_URI);
+		when(adminService.getGlobalProperty(GP_SANTE_CLIENT_ID)).thenReturn(SANTE_CLIENT_ID);
+		when(adminService.getGlobalProperty(GP_SANTE_CLIENT_SECRET)).thenReturn(SANTE_CLIENT_SECRET);
+		when(MpiUtils.getGlobalPropertyValue(GP_UUID_SYSTEM)).thenReturn(UUID_SYSTEM);
+		mpiContext.setAuthenticationType(AUTHENTICATION_TYPE.CERTIFICATE);
+		
+	}
+	
+	@Test
+	public void getPatient_shouldRetrievePatientInOpenCR() throws Exception {
+		
+		final String patientUuid = "patient-uuid-open-cr";
+		String baseUrl = "https://localhost:8080";
+		Map<String, Object> patientResponse = new HashMap<>();
+		patientResponse.put(FIELD_ACTIVE, true);
+		patientResponse.put(OPENMRS_UUID, patientUuid);
+		
+		when(MpiContext.initIfNecessary()).thenReturn(mpiContext);
+		when(mpiContext.getAuthenticationType()).thenReturn(AUTHENTICATION_TYPE.CERTIFICATE);
+		when(mpiContext.getServerBaseUrl()).thenReturn(baseUrl);
+		when(mockMpiHttpClient.getPatient(patientUuid)).thenReturn(patientResponse);
+		Map<String, Object> openCRResponse = mockMpiHttpClient.getPatient(patientUuid);
+		
+		assertEquals(openCRResponse.get(FIELD_ACTIVE), patientResponse.get(FIELD_ACTIVE));
```

Comment:
I don't see where you are verifying that the patient request was submitted to OpenCR